get_file_copy.go

```diff
@@ -27,3 +28,32 @@ func Copy(ctx context.Context, dst io.Writer, src io.Reader) (int64, error) {
 		}
 	}))
 }
+
+// CopyReader copies from an io.Reader into a file, using umask to create the dst file
+func CopyReader(dst string, src io.Reader, umask os.FileMode) error {
+	dstF, err := os.OpenFile(dst, os.O_RDWR|os.O_CREATE|os.O_TRUNC, mode(0666, umask))
```

Comment:
Since this accepts a mode, can we make CopyReader take the mode instead of the umask and skip the Chmod we do after CopyReader? Would save a ton of syscalls and filesystem metadata writes on large archives with lots of small files.

Reply:
Good suggestion, which highlights a quirk of `libc` that is inherited by golang's `os.*`. The file mode can't be used to setuid/gid bits on creation, those bits must be explicitly set in a `chmod` call. This is lightly documented but widespread behavior.
See https://github.com/golang/go/issues/25539

The most future/platform proof way to manage umask is probably to calculate the mode ourselves, although on linux and bsds today its unnecessary duplication, libc will strip setuid/setgid bits on linux, and all special permissions on bsds

Reply:
this change is backed out now, we need to use chmod directly to preserve the permissions from the archive